src/clr-angular/data/datagrid/datagrid-column.ts

```diff
@@ -75,34 +90,67 @@ import { DetailService } from './providers/detail.service';
           <span class="datagrid-column-title" *ngIf="!sortable">
               <ng-container *ngTemplateOutlet="columnTitle"></ng-container>
           </span>
-
-          <clr-dg-column-separator *ngIf="showSeparator"></clr-dg-column-separator>
+      </div>
+      </div>
+      <clr-dg-column-separator></clr-dg-column-separator>
+      <div class="datagrid-column-droppable" 
+      clrDroppable 
+      [clrGroup]="columnsGroupId"
+      (clrDragStart)="_markForCheck = true; closeFilterWhenDragStart()"
+      (clrDrop)="requestReorder($event);">
+        <div class="datagrid-column-drop-line"></div>
       </div>
     `,
   providers: [ClrPopoverPositionService, ClrPopoverEventsService, ClrPopoverToggleService],
   host: {
     '[class.datagrid-column]': 'true',
+    '[class.datagrid-column-drag-mode]': 'inDragMode',
+    '[class.datagrid-column-reorder-drop]': 'inReorderDrop',
+    '[class.datagrid-column-reorder-shift]': 'inReorderShift',
     '[attr.aria-sort]': 'ariaSort',
     role: 'columnheader',
   },
+  animations: [
+    trigger('reorderDropAnimation', [
+      transition(`${DROP_ANIM_STATE.INACTIVE} => ${DROP_ANIM_STATE.ACTIVE}`, [
+        style({
+          position: 'fixed',
+          width: '{{width}}px',
+          height: '{{height}}px',
+          top: '{{fromTop}}px',
+          left: '{{fromLeft}}px',
+        }),
+        animate('0.2s 0.1s ease-in-out', style({ top: '{{toTop}}px', left: '{{toLeft}}px' })),
+      ]),
+    ]),
+    trigger('reorderShiftAnimation', [
+      transition(`${SHIFT_ANIM_STATE.INACTIVE} => ${SHIFT_ANIM_STATE.ACTIVE}`, [
+        style({ position: 'relative', left: '{{fromLeft}}px' }),
+        animate('0.2s ease-in-out', style({ left: '0px' })),
+      ]),
+    ]),
+  ],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ClrDatagridColumn<T = any> extends DatagridFilterRegistrar<T, ClrDatagridFilterInterface<T>>
-  implements OnDestroy, OnInit {
+  implements ViewAccessor, OnDestroy, OnInit {
   constructor(
     private _sort: Sort<T>,
     filters: FiltersProvider<T>,
     private vcr: ViewContainerRef,
-    private detailService: DetailService,
     private changeDetectorRef: ChangeDetectorRef,
-    public commonStrings: ClrCommonStringsService
+    public commonStrings: ClrCommonStringsService,
+    private columnReorderService: ColumnReorderService,
+    private filterPopoverToggleService: ClrPopoverToggleService,
+    private domAdapter: DomAdapter
   ) {
     super(filters);
     this.subscriptions.push(this.listenForSortingChanges());
-    this.subscriptions.push(this.listenForDetailPaneChanges());
+    this.subscriptions.push(this.listenForReorderAnimRequest());
   }
 
-  public showSeparator = true;
+  // TODO: remove once OnPush change detection strategy is removed
+  _markForCheck = false;
```

Comment:
I know some things were removed, is it still a problem?

Reply:
This is no longer a problem. It will be removed

Reply:
It should be removed when I rebase the branch as we have already got rid of OnPush in master.